packages/e2e/src/helper.ts

```diff
@@ -102,13 +104,17 @@ export const setupAll = async ({
         noInitWarn: true,
       })
 
+      const substrateClient = createClient(getWsProvider(`ws://localhost:${port}`))
```

Comment:
I am not sure how much overhead is there to create a papi client because this is used by all tests and papi client will only be used by some tests. Maybe add an option to create papi client when needed?

Reply:
There shouldn't be too much overhead other than a new WS connection opened - `@polkadot-api/substrate-client` doesn't do any operations by its own.

I can move it to a separate function or add the option flag if it's still a concern 👍

Reply:
I've been adding the transaction_v1 and when writing the tests I found out it would be great to actually have the v15 metadata, because then we can use a higher-level library to track transactions and check for success, invalid and failed.

I'll take this chance to put it as a separate env that only gets instantiated in the new RPC spec tests (on this PR)

Reply:
can you put this back to draft and make it ready when all the work is done so I know when it is a good time to review again?

Reply:
Sorry I missed this - It's now ready 👍